saaj-ri/src/main/java/com/sun/xml/messaging/saaj/util/transform/EfficientStreamingTransformer.java

```diff
@@ -73,16 +74,33 @@
      * Underlying FI DOM serializer.
      */
     private Object m_fiDOMDocumentSerializer = null;
+
+    private static final AtomicBoolean TO_BE_LOGGED = new AtomicBoolean(true);
```

Comment:
what is the point of adding a property which one cannot set/update externally?

Reply:
As Brain explained below, this is to get this log only once  instead of each time.

Reply:
for two different apps deployed on the server where the library is included on the server and each app uses different transformer implementation - which one will be logged and which won't be? In any case - this logic seems to be out of the scope of the issue being fixed here. What I'd rather see is a bug and a solution consistent with the rest of the stack (currently ~3-5 other repos; ideally shared logic/code should be consolidated to one or two repos and solution applied there)

Reply:
There is only one application should be logged, and only the application which uses xalan implementation will hit the line to log the warning message. IMO, this is better solution to skip the duplicate warning message to improve the performance when the SOAPMessage.saveChange() is called to create  a soap message like:
```
      MessageFactory factory = MessageFactory.newInstance();
      SOAPMessage message = factory.createMessage();
      message.getSOAPPart().setContent((Source) stringMsg);
      message.saveChanges();
```